support/sys-socket.c

```diff
@@ -0,0 +1,139 @@
+/*
+ * <sys/socket.h> wrapper functions.
+ *
+ * Authors:
+ *   Steffen Kiess (s-kiess@web.de)
+ *
+ * Copyright (C) 2015 Steffen Kiess
+ */
+
+#include <sys/socket.h>
+#include <netinet/in.h>
+#include <sys/un.h>
+
+#include <stddef.h>
+
+#include "map.h"
+#include "mph.h"
+
+G_BEGIN_DECLS
+
+int
+Mono_Posix_Syscall_socketpair (int domain, int type, int protocol, int* sv0, int* sv1)
```

Comment:
> The reason I hade the (out int, out int) overload is because pipe() also has them, but I don't mind removing them for socketpair().

"Mistakes were made." :-)

I'm now of the opinion that the fewer overloads the better, as there's less code to screw up that way. ;-)

(I had a different opinion at the time I wrote `pipe()`.)

Furthermore, the reason `pipe()` has the overload with `out` parameters is simple: *I hate the array*. Quick, without consulting the documentation, which index is for reading vs. writing?

`out` parameters also mix nicely with C#4 named parameters, allowing for straightforward *names* instead of *number indexes*:

    int read, write;
    Syscall.pipe (reading: out read, writing: out write);

I shouldn't have written `pipe(int[])`.

With that in mind, `socketpair()` shouldn't repeat the mistakes of `pipe()`: it should pick one overload, not two, and stick with it.

Which overload? I'd still argue for the `out` parameters (readability!), but I can be convinced otherwise. :-)